### ISSUE TITLE
Error

### ISSUE THREAD

**Issue by Joan1211** [c17hawke/chatGPT-Telegram-bot#1]
Hello. After following the tutorial, i have this error:

I have this error:

(env) me@chatGT2:~/chatGPT-Telegram-bot-main$ python3 src/chatgpt.py
Traceback (most recent call last):
File " /home/me/chatPT-Telegram-bot-main/src/chatgpt.py", line 29, in <module>
class Reference:
File " /home/me/chatGPT-Telegram-bot-main/src/chatgpt.py",line 33, in Reference
raise NotImplementedError
NotImplementedError

Any help? Thanks

**Comment by Joan1211** [comment]
Edit:
After reviewing the video, it appears that the given code lacks several entries.

There's a necessity to fill in these missing parts manually in order to address all the _'raise not implemented error'_ issues.

Furthermore, the creation of a .env file for the keys isn't functioning as expected. 
You will need to update few things inside chatgpt.py.
to direct the code to read the .env for openAI key.
to modify the TELEGRAM-BOT-TOKEN into TOKEN in the env.

To sum up, 
I am not sure the relevance of this tutorial to the provided code.
It seems like it was intended for a different script.

**Comment by c17hawke** [comment]
Hi @Joan1211

Thanks for pointing this out, actually the updated code was not pushed to the repository, so apologies for that. 
I have updated the code now for your reference. 
Let me know if you face any other issues!

Thanks!

**Comment by Joan1211** [comment]
Thank you very much for this help. I managed to make it work. but now i will try with your updated files !
